src/Umbraco.Cms.ManagementApi/Filters/RequireRuntimeLevelAttribute.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.ManagementApi.Filters;
+
+public class RequireRuntimeLevelAttribute : ActionFilterAttribute
+{
+    private readonly RuntimeLevel _requiredRuntimeLevel;
+
+    public RequireRuntimeLevelAttribute(RuntimeLevel requiredRuntimeLevel)
+    {
+        _requiredRuntimeLevel = requiredRuntimeLevel;
+    }
+
+    public override void OnActionExecuting(ActionExecutingContext context)
+    {
+        IRuntimeState runtimeState = context.HttpContext.RequestServices.GetRequiredService<IRuntimeState>();
+
+        // We're not in the expected runtime level, so we need to short circuit
+        if (runtimeState.Level != _requiredRuntimeLevel)
```

Comment:
Consider early exit instead of nesting - `if (runtimeState.Level == _requiredRuntimeLevel) { return; }`